include/nbl/system/IThreadHandler.h

```diff
@@ -0,0 +1,103 @@
+#ifndef __NBL_I_THREAD_HANDLER_H_INCLUDED__
+#define __NBL_I_THREAD_HANDLER_H_INCLUDED__
+
+#include <mutex>
+#include <condition_variable>
+#include <thread>
+
+namespace nbl {
+namespace system
+{
+
+// Usage:
+/*
+* class MyThreadHandler : public IThreadHandler<SomeInternalStateType> { .... };
+* 
+* MyThreadHandler handler;
+* std::thread thread(&MyThreadHandler::thread, &handler);
+* //...
+* //... communicate with the thread using your methods (see note at the end of this section)), thread will sleep until wakeupPredicate() returns true
+* //...
+* handler.terminate(thread);
+* After this handler can be safely destroyed.
+* Every method playing around with object's state shared with the thread must begin with line: `auto raii_handler = createRAIIDisptachHandler();`!
```

Comment:
well it should just `auto raii_lock = createLock()` to play around with internal state.

if you want to modify state and make the listening thread wake up then yes `auto raii_handler = createRAIIDisptachHandler();`

Reply:
yea, it is like that, bad docs, ill fix the docs